lib/ansible/parsing/ajson.py

```diff
@@ -36,6 +36,10 @@ def _decode_map(self, value):
             for k in value:
                 if isinstance(value[k], Mapping):
                     value[k] = self._decode_map(value[k])
+                elif isinstance(value[k], list):
+                    for i in range(len(value[k])):
```

Comment:
i'm not sure you need to traverse the list as the json decoder should already be doing that

Reply:
It's not already doing this, I can file an issue to give more details if needed.

Tomorrow morning I'll see about re-testing with @sivel's diff.

Reply:
you are probably correct, i remember going over the traversal with default, but i didn't check if it did the same here.

Reply:
@sivel confirmed with the noted steps, from top of my inventory examples repo, run:

```
ansible-playbook debugging/hostvars_print.yml -i scripts/vault/bug2226/single.py --ask-vault-pass
```

(provide foobar as secret)

After applying your diff the content in the list under the "baz" hostvar is decrypted, but it is not in devel.

👍 to that as far as I'm concerned. Would you like to make it into a PR?

Reply:
@AlanCoding sorry for the delay, I'll try to get a PR together soon.